tiledb/ml/readers/types.py

```diff
@@ -42,15 +48,48 @@ def __post_init__(self) -> None:
             all_dims[0], all_dims[key_dim_index] = all_dims[key_dim_index], all_dims[0]
             ned[0], ned[key_dim_index] = ned[key_dim_index], ned[0]
 
-        object.__setattr__(
-            self,
-            "_tensor_schema_kwargs",
-            dict(
-                array=self.array,
-                fields=tuple(final_fields),
-                key_dim_index=key_dim_index,
-                ned=tuple(ned),
-                all_dims=tuple(all_dims),
-                query_kwargs={"attrs": tuple(attrs), "dims": tuple(dims)},
-            ),
+        tensor_schema_kwargs = dict(
+            _array=self.array,
+            _key_dim_index=key_dim_index,
+            _fields=tuple(final_fields),
+            _all_dims=tuple(all_dims),
+            _ned=tuple(ned),
+            _query_kwargs={"attrs": tuple(attrs), "dims": tuple(dims)},
         )
+        object.__setattr__(self, "_tensor_schema_kwargs", tensor_schema_kwargs)
+
+    def to_tensor_schema(
+        self,
+        transforms: Mapping[TensorKind, Union[Callable[[Any], Any], bool]] = {},
+    ) -> TensorSchema[Any]:
+        """
+        Create a TensorSchema from an ArrayParams instance.
+
+        :param transforms: A mapping of `TensorKind`s to transformation callables.
+            If `array_params.tensor_kind` (or the inferred tensor_kind for `array_params`)
+            has a callable value in `transforms`, the returned `TensorSchema` will map
+            each tensor yielded by its `iter_tensors` method with this callable.
+
+            A value in transforms may also be a boolean value:
```

Comment:
Could you please explain even further what exactly does the boolean option means? It sounds to me counter intuitive to pass an argument `transforms=True` that leads to `no transformation will be applied`

Reply:
So the main use case is `False` to denote that this kind is not supported. `True` is added for completeness (it would be odd to allow only `False`) and is equivalent to not having an entry for this key. It can be thought of as equivalent to the `lambda x: x` identity mapper. 

Initially I used the [`NotImplemented`](https://docs.python.org/3/library/constants.html#NotImplemented) singleton for `False` and nothing for True that is more intuitive but this [doesn't currently play well with mypy](https://github.com/python/mypy/issues/4791).